Description:
entitymanager/src/main/java/org/hibernate/ejb/criteria/path/AbstractPathImpl.java:[194,39]
invalid inferred types for M; inferred type does not conform to declared bound(s)
inferred: java.util.Map<K,V>
bound(s): java.util.Map<capture#829 of ?,capture#332 of ?>

I suspect this failure is because IcedTea 1.8.2 got more
strict with type inference and generics code. I say this
since I can rebuild this package without problems with
1.8.1. Thus, this FTBFS is present in sid and squeeze.

This bug seems to be very similar to #602362.
The problematic code involves the usage of Java generics and
since I'm not very experienced with that kind of code I'm
asking for help.

Bug: http://bugs.debian.org/606295
Author: Damien Raude-Morvan <drazzib@debian.org>
Last-Update: 2010-12-28
Forwarded: yes
--- a/entitymanager/src/main/java/org/hibernate/ejb/criteria/path/AbstractPathImpl.java
+++ b/entitymanager/src/main/java/org/hibernate/ejb/criteria/path/AbstractPathImpl.java
@@ -191,10 +191,10 @@
 		if ( attribute.isCollection() ) {
 			final PluralAttribute<X,Y,?> pluralAttribute = (PluralAttribute<X,Y,?>) attribute;
 			if ( PluralAttribute.CollectionType.MAP.equals( pluralAttribute.getCollectionType() ) ) {
-				return (PluralAttributePath<Y>) get( (MapAttribute<X,?,?>) pluralAttribute );
+				return (PluralAttributePath<Y>) this.<Object,Object,Map<Object, Object>>get( (MapAttribute) pluralAttribute );
 			}
 			else {
-				return (PluralAttributePath<Y>) get( (PluralAttribute) pluralAttribute );
+				return (PluralAttributePath<Y>) this.get( (PluralAttribute) pluralAttribute );
 			}
 		}
 		else {
